### ISSUE TITLE
what are dataset_means and what is their impact?

### ISSUE THREAD

**Issue by wisdomk** [amdegroot/ssd.pytorch#314]
I would like to understand what is the purpose of [dataset_means](https://github.com/amdegroot/ssd.pytorch/blob/master/eval.py#L75) in '[eval.py](https://github.com/amdegroot/ssd.pytorch/blob/master/eval.py#L75)' and how they affect the results as whenever I change the numbers the end results AP and mean AP will change as well!

could somebody please explain a little bit and tell their impact on new trained dataset?

**Comment by pinkerltm** [comment]
This is also done in training the network. Generally speaking: Subtracting the mean of a dataset is done to center the whole dataset around zero. Additionaly you would divide the dataset by its standard deviation (not done in this project) which scales all values between -1 and +1. This is called normalization and helps to regularize gradients while using higher learning rates. 

If you introduce "Batch normalization" to you pytorch model (https://towardsdatascience.com/batch-normalization-in-neural-networks-1ac91516821c) you could also skip this preprocessing step as it is done minibatch wise anyway.

There would be also an increase-signal-to-noise ratio effect if you subtract the mean of all images of a certain camera for example which would then implicitly remove noise caused by this certain camera.

**Comment by Thome** [comment]
I'm still not sure what exactly that means. What if I had my own custom dataset? How would I find these means? Or are they the same regardless?

**Comment by zhaoedf** [comment]
> I'm still not sure what exactly that means. What if I had my own custom dataset? How would I find these means? Or are they the same regardless?

want to know this too!  keep wondering that these numbers should remain what they are or I should manually make them adapted to my own datasets?
